assets/src/blocks/components/Lightbox/setupLightboxForImages.js

```diff
@@ -1,17 +1,74 @@
 import {Lightbox} from './Lightbox';
 import {createRoot} from 'react-dom/client';
 
+/**
+ * Extracts the WordPress image ID from an element's class list.
+ *
+ * @param {HTMLElement} el - The DOM element to inspect.
+ * @return {number|null} The image ID if found, otherwise null.
+ */
+const getImageIdFromClass = el => {
+  for (const cls of el.classList) {
+    const match = cls.match(/^wp-image-(\d+)$/);
+    if (match) {
+      return parseInt(match[1], 10);
+    }
+  }
+  return null;
+};
+
+/**
+ * Fetches metadata for a WordPress image by its DOM element.
+ *
+ * @param {HTMLImageElement} imgEl - The image element.
+ * @return {Promise<{url: string, width: number|null, height: number|null}|null>}
+ *   An object containing the image URL and dimensions, or null if not found.
+ */
+const getImageMeta = async imgEl => {
+  const imgId = getImageIdFromClass(imgEl);
+  if (!imgId) { return null; }
+
+  try {
+    const response = await fetch(`/wp-json/wp/v2/media/${imgId}`);
+    if (!response.ok) { throw new Error('Failed to fetch media info'); }
```

Comment:
Thinking loud, do we really need to `throw` en `Error`? What if we log that as an error instead?